languages/rust/exercises/concept/tuples/.meta/example.rs

```diff
@@ -0,0 +1,66 @@
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum Unit {
+    Millilitre,
+    UsCup,
+    UsFluidOnce,
+    UsTeaspoon,
+    UsTablespoon,
+}
+
+pub fn get_volume(volume: &(Unit, f32)) -> f32 {
+    volume.1
+}
+
+pub fn to_cup(volume: &(Unit, f32)) -> (Unit, f32) {
+    match volume {
+        (Unit::Millilitre, volume)  => {
+            let converted = volume * (1.0/240.0);
+            (Unit::UsCup, format!("{:.2}", converted).parse::<f32>().unwrap())
+        }
+        (_, _) => unimplemented!("Not supported yet")
+    }
+}
+pub fn to_millilitres(volume: &(Unit, f32)) -> (Unit, f32) {
+    match volume {
+        (Unit::UsCup, volume)  => {
+            let converted = volume * 240.0;
+            (Unit::Millilitre, format!("{:.2}", converted).parse::<f32>().unwrap())
+        }
+        (_, _) => unimplemented!("Not supported yet")
+    }
+}
+pub fn convert(volume: &(Unit, f32), destination: Unit) -> (Unit, f32) {
```

Comment:
```suggestion
pub fn convert(volume: (Unit, f32), destination: Unit) -> (Unit, f32) {
```

It's not normally good practice to use references to `Copy` types in function signatures. Tuples for which all members are `Copy` are also `Copy`. To demonstrate that property, remove the reference in this function's signature, and try this:

```rust
#[test]
fn tuples_are_copy() {
    let input = (Unit::Millilitre, 15.0);
    assert_eq!(
        convert(input, Unit::UsTablespoon),
        (Unit::UsTablespoon, 1.0)
    );
    assert_eq!(
        convert(input, Unit::UsTeaspoon),
        (Unit::UsTeaspoon, 3.0)
    );
}
```